src/model_signing/manifest.py

```diff
@@ -466,3 +468,51 @@ def serialization_type(self) -> dict[str, Any]:
         manifest so that signature verification can use the same method.
         """
         return self._serialization_type.serialization_parameters
+
+    @classmethod
+    def from_signature(cls, signature_path: pathlib.Path) -> Self:
```

Comment:
This should probably call the verification path first and then use https://github.com/sigstore/model-transparency/blob/992a56ba2c4f84727729498e5dbd2c16ad5b0d7c/src/model_signing/_signing/signing.py#L353 to get the manifest directly without duplication.

Reply:
Ok that's what I tried to do! Will refresh the review for you to take a second look at it